cli/listener/src/main.rs

```diff
@@ -570,11 +570,41 @@ fn get_success_response(interval_prices: IntervalPrices) -> ResponseBox {
     .boxed()
 }
 
+/// Get a header with `Expires` key.
+/// If the end date for the query is far in the past we set it to expire in 1
+/// year. However, if the end date is in the same epoch, we set it to expire in
+/// the approximate time the next epoch begins.
+fn get_expires_header(
+    now: chrono::DateTime<chrono::Utc>,
+    current_slot: Slot,
+    interval_prices: &IntervalPrices,
+) -> Header {
+    let epoch_sched = EpochSchedule::without_warmup();
+    let (epoch_for_current_slot, _) = epoch_sched.get_epoch_and_slot_index(current_slot);
+
+    // Set the validity of the query until the next epoch
+    let expires_date = if epoch_for_current_slot == interval_prices.end_epoch {
+        let first_slot_next_epoch = epoch_sched.get_first_slot_in_epoch(epoch_for_current_slot + 1);
+        // Does not underflow because `first_slot_next_epoch <= current_slot`.
+        let slots_diff = first_slot_next_epoch - current_slot;
+        // Estimate for how many milliseconds the response is valid.
+        let ms_diff = slots_diff * SLOT_MS;
+        now + chrono::Duration::milliseconds(ms_diff as i64)
+    } else {
+        // 1 year duration if the end is in the past.
+        now + chrono::Duration::days(365)
```

Comment:
In theory this should be fine, but in practice, if we mess something up, one year is a long time to have to wait for the bad data to go away. (I think there is an option to flush the cache in Cloudflare, but it will be a hassle to get it executed.) In terms of load, it doesn’t matter *that* much when a CDN can save us 50 requests a year, the case where the CDN helps us is when we get thousands of requests a second. So we might also cache it for just 7 days. Then we get ~1 request per week (for a given url) which is still virtually nothing.